multipart-form-writer/multipart_form_writer.ml

```diff
@@ -62,7 +76,11 @@ let open_file path =
   |> Lwt_io.open_file ~mode:Lwt_io.Input
   >|= read_while_not_empty
   >|= Lwt_stream.from
-  |> Lwt_result.ok
+  >>= fun file_stream -> safe_file_size path
+  >|= fun file_size ->
+  match file_size with
+  | Ok file_size -> Ok (Progressbar.progress_bar ~size:file_size file_stream)
```

Comment:
Would it be possible to avoid having a dependency from `Multipart_form_writer` to `Progressbar`?  This is the stuff you would only expect a CLI module like `Cs_api_cli` to depend on.

I'm afraid it would require some plumbing so that `Cs_api_cli` can provide a callback to `Cs_api_io` which would apply it to the stream, so it might not be worth it unfortunately.

Reply:
So this is a good idea, but trying to rewrite the package, I ran into a lot of problems that were very similar to the ones I am currently trying to solve in the multipart-form-data package. So I suggest we keep the structure as is for now, and rewrite it properly once the multipart-form-data package is integrated.

Reply:
OK, fair enough.